pip/index.py

```diff
@@ -309,7 +309,7 @@ def mkurl_pypi_url(url):
                 loc = loc + '/'
             return loc
 
-        url_name = req.url_name
+        project_url_name = urllib_parse.quote(project_name.lower())
```

Comment:
Well, it was the only place that used `req.url_name`, so I found it was simpler to drop it:
https://github.com/pypa/pip/pull/2614/files#diff-297395da5417032dc5d0738a1fc7e136L321